### ISSUE TITLE
No video on macOS 10.15

### ISSUE THREAD

**Issue by LordMurillo** [packagesdev/savehollywood#15]
After upgrading to macOS Catalina (10.15), it shows me "no videos". When I check the preferences, the list of directories is empty, and if I try to add a new directory, when I click on "+", it does nothing.
SaveHollywood version is 2.5 (9)

**Comment by nextgenoperations** [comment]
See my post on the other thread referring to 10.15:

Ok, I have it working in 10.15 now... the trick was (due to the sandboxing I believe) was to put the video file in the actual folder where the plist resides.... (new folder in 10.15).... which is:

/Users/XXXXX/Library/Containers/com.apple.ScreenSaver.Engine.legacyScreenSaver/Data/Library/Preferences/ByHost

Once I placed the video in this folder, I 'dragged and dropped' it into the config for savehollywood and it works fine...

Perhaps not a solution, but at least a workaround...

@packagesdev I am guessing this it most likely a permissions issue with the new sandboxing for 3rd party screeen savers...

Adam

**Comment by LordMurillo** [comment]
I don't quite understand what you're saying.
Do I have to copy the videos on the path /Users/XXXXX/Library/Containers/com.apple.ScreenSaver.Engine.legacyScreenSaver/Data/Library/Preferences/ByHost?

**Comment by nextgenoperations** [comment]
Yes.

**Comment by LordMurillo** [comment]
It doesn't work for me.

**Comment by nextgenoperations** [comment]
Did you place the video In the  folder ?

Once done, did you ‘drag and drop’ the video into the savehollywood video field (the + button doesn’t work)

This is what I did and others have managed to get it to work ....

I am not the developer so anything more I can’t really help...

Adam

**Comment by LordMurillo** [comment]
It's working!
 I didn't understand you correctly
Thank you very much

**Comment by nextgenoperations** [comment]
Good news !!

**Comment by hclyatt** [comment]
> See my post on the other thread referring to 10.15:
> 
> Ok, I have it working in 10.15 now... the trick was (due to the sandboxing I believe) was to put the video file in the actual folder where the plist resides.... (new folder in 10.15).... which is:
> 
> /Users/XXXXX/Library/Containers/com.apple.ScreenSaver.Engine.legacyScreenSaver/Data/Library/Preferences/ByHost
> 
> Once I placed the video in this folder, I 'dragged and dropped' it into the config for savehollywood and it works fine...
> 
> Perhaps not a solution, but at least a workaround...
> 
> @packagesdev I am guessing this it most likely a permissions issue with the new sandboxing for 3rd party screeen savers...
> 
> Adam

I stored my videos in: 

/Users/XXXXX/Library/Containers/com.apple.ScreenSaver.Engine.legacyScreenSaver/Data/Library/ApplicationSupport/SaveHollywood

I had to create the folder SaveHollywood myself. I feel this may be a more appropriate location for the files than Preferences and is the same location Aerial uses. 

I was then able to drag and drop the files into the SaveHollywood screensaver preferences as suggested and everything works.

Hank

**Comment by nextgenoperations** [comment]
> > See my post on the other thread referring to 10.15:
> > Ok, I have it working in 10.15 now... the trick was (due to the sandboxing I believe) was to put the video file in the actual folder where the plist resides.... (new folder in 10.15).... which is:
> > /Users/XXXXX/Library/Containers/com.apple.ScreenSaver.Engine.legacyScreenSaver/Data/Library/Preferences/ByHost
> > Once I placed the video in this folder, I 'dragged and dropped' it into the config for savehollywood and it works fine...
> > Perhaps not a solution, but at least a workaround...
> > @packagesdev I am guessing this it most likely a permissions issue with the new sandboxing for 3rd party screeen savers...
> > Adam
> 
> I stored my videos in:
> 
> /Users/XXXXX/Library/Containers/com.apple.ScreenSaver.Engine.legacyScreenSaver/Data/Library/ApplicationSupport/SaveHollywood
> 
> I had to create the folder SaveHollywood myself. I feel this may be a more appropriate location for the files than Preferences and is the same location Aerial uses.
> 
> I was then able to drag and drop the files into the SaveHollywood screensaver preferences as suggested and everything works.
> 
> Hank

Agreed that this location is probably better, after some reading on the Apple Dev site, anywhere under the screensaver.legacy container is valid as savehollywood will be able to read the file...

A.

**Comment by Lastofthefirst** [comment]
Thanks, this worked for me.

**Comment by gjw891260896** [comment]
Not support 4k video？

**Comment by AraHaan** [comment]
Not working on MacOS 11.3 Beta?

**Comment by 153957** [comment]
Note that Aerial also supports showing your own local videos, and works on Monterey: https://aerialscreensaver.github.io

**Comment by MDanihy** [comment]
> Note that Aerial also supports showing your own local videos, and works on Monterey: https://aerialscreensaver.github.io

Not well, I get crashing and just black screens.  Error loading file errors.

**Comment by AraHaan** [comment]
I wonder if something like this can be made for Windows and Linux as well for fun 🤔.

**Comment by 153957** [comment]
> I wonder if something like this can be made for Windows and Linux as well for fun 🤔.

(From the Aerial readme)
Windows – https://github.com/OrangeJedi/Aerial
Linux – https://github.com/graysky2/xscreensaver-aerial/

**Comment by fakesamgregory** [comment]
I just tested moving it outside an iCloud folder and it works. It won't work Inside an iCloud synced folder by my observations.
